src/parsers.rs

```diff
@@ -204,8 +196,25 @@ fn time_second(i: &[u8]) -> IResult<&[u8], u32> {
     n_digit_in_range(i, 2, 0..=60)
 }
 
-fn time_millisecond(fraction: f32) -> u32 {
-    (1000.0 * fraction) as u32
+fn time_millisecond(i: &[u8]) -> IResult<&[u8], u32> {
+    let (i, mut digits) = take_while(is_digit)(i)?;
+    let mut l = digits.len();
+    if l > 3 {
+        digits = digits.get(0..3).unwrap();
```

Comment:
What happens if there's more digits? Will those digits now just be dropped?

Reply:
Yes, digits beyond the first 3 are dropped.

This is consistent with the behavior of the previous implementation, as casting from a floating point number to an integer rounds towards zero.

Reply:
Right. All good  then.